akka-http-caching/src/main/scala/akka/http/caching/LfuCache.scala

```diff
@@ -105,7 +105,21 @@ private[caching] class LfuCache[K, V](val store: AsyncCache[K, V]) extends Cache
 
   def apply(key: K, genValue: () => Future[V]): Future[V] = store.get(key, toJavaMappingFunction[K, V](genValue)).toScala
 
-  def getOrLoad(key: K, loadValue: K => Future[V]) = store.get(key, toJavaMappingFunction[K, V](loadValue)).toScala
+  def put(key: K, mayBeValue: Future[V])(implicit ex: ExecutionContext): Future[V] = {
+    val previouslyCacheValue = Option(store.getIfPresent(key))
+
+    previouslyCacheValue match {
+      case None =>
+        store.put(key, toJava(mayBeValue).toCompletableFuture)
+        mayBeValue
+      case _ => mayBeValue.map { value =>
+        store.put(key, toJava(Future.successful(value)).toCompletableFuture)
```

Comment:
Isn't that prone to race conditions if multiple `put`s are ongoing for the same key? What may be surprising is that after a sequence of multiple `put`s to the same key the content of the cache might not be the result of the last call.

It's a cache so maybe we shouldn't care? Any opinion, @raboof?

Reply:
If we agree that the race conditions are ok then we should add a note about that to the docs as well.

Reply:
I think they are fine for now indeed.

Keeping track of multiple parallel 'in-flight' `put`s might be an interesting optimization, but that's still possible to add later without breaking the API.